dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java

```diff
@@ -223,44 +229,80 @@ private void connect(String lastEventId) {
+            if (status != 200 || !MediaType.SERVER_SENT_EVENTS.equals(contentType)) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.log(Level.FINEST, "Received " + status + " Content-Type=" + contentType);
+                }
+                final Response fResponse = response;
+                Throwable t;
+                if (!MediaType.SERVER_SENT_EVENTS.equals(contentType)) {
+                    t = new WebApplicationException("Unexpected Content-Type in response", response);
+                } else {
+                    t = AccessController.doPrivileged((PrivilegedExceptionAction<Throwable>)() -> {
+                        Class<? extends Throwable> throwableClass = (Class<? extends Throwable>) 
+                                        ExceptionUtils.getWebApplicationExceptionClass(fResponse, WebApplicationException.class);
+                        Constructor<? extends Throwable> ctor;
+                        try {
+                            ctor = throwableClass.getConstructor(Response.class);
+                        } catch (NoSuchMethodException ex) {
+                            ctor = null;
+                        }
+                        return ctor == null ? throwableClass.newInstance() : ctor.newInstance(fResponse);
+                    });
+                }
+
+                delegate.onError(t);
+                delegate.onComplete();
+                response.close();
+                return;
+            }
             // Should not happen but if close() was called from another thread, we could
             // end up there.
             if (state.get() == SseSourceState.CLOSED) {
                 LOG.fine("SSE connection to " + target.getUri() + " has been closed already");
                 response.close();
                 return;
             }
-            
-            final Endpoint endpoint = WebClient.getConfig(target).getEndpoint();
+
             // Create new processor if this is the first time or the old one has been closed 
             if (processor == null || processor.isClosed()) {
+                final Endpoint endpoint = WebClient.getConfig(target).getEndpoint();
                 LOG.fine("Creating new instance of SSE event processor ...");
                 processor = new InboundSseEventProcessor(endpoint, delegate);
             }
-            
+
             // Start consuming events
             processor.run(response);
             LOG.fine("SSE event processor has been started ...");
-            
+
             if (!state.compareAndSet(SseSourceState.CONNECTING, SseSourceState.OPEN)) {
                 throw new IllegalStateException("The SseEventSource is already in " + state.get() + " state");
             }
-            
+
             LOG.fine("Successfuly opened SSE connection to " + target.getUri());
         } catch (final Exception ex) {
+            LOG.log(Level.WARNING, "ANDY caught exception in connect(...)", ex); // TODO REMOVE THIS!
```

Comment:
I assume you are ok with delivering this with "Andy" in it beyond the TODO eye catcher?

Reply:
good catch - I'll remove this - and the other occurrences.  :)